### PR TITLE
add demos for cryptographic functions with and without

### PR DESCRIPTION
**Pull request by jpbland1**
imported keys

### CODE REVIEW

File: demo/client/wh_demo_client_crypto.c [hunk 1]
```diff
@@ -1,7 +1,1347 @@
...
+    whKeyId keyIdPriv1 = WOLFHSM_KEYID_ERASED;
+    whKeyId keyIdPub1 = WOLFHSM_KEYID_ERASED;
+    whKeyId keyIdPriv2 = WOLFHSM_KEYID_ERASED;
+    whKeyId keyIdPub2 = WOLFHSM_KEYID_ERASED;
+    word32 outLen;
+    word32 sigLen;
+    char privKeyFileOne[] = "../../../demo/certs/ecc-private-one.raw";
+    char pubKeyFileOne[] = "../../../demo/certs/ecc-public-one.raw";
+    char privKeyFileTwo[] = "../../../demo/certs/ecc-private-two.raw";
+    char pubKeyFileTwo[] = "../../../demo/certs/ecc-public-two.raw";
```
Comment (bigbrett) [inline demo/client/wh_demo_client_crypto.c:677]: same comment as above re: Alice and Bob

File: demo/client/wh_demo_client_crypto.c [hunk 2]
```diff
@@ -1,7 +1,1347 @@
...
+        printf("AES GCM doesn't match after decryption\n");
+        ret = -1;
+        goto exit;
+    }
+    printf("AES GCM matches after decryption\n");
+exit:
+    return ret;
+}
+
+int wh_DemoClient_CryptoAesGcmImport(whClientContext* clientContext)
```
Comment (bigbrett) [inline demo/client/wh_demo_client_crypto.c:1136]: I think it is slightly clearer to the reader if you cache the key first thing, then indicate the subsequent AES operations (including init) can happen at any point later, even after a power cycle, hence the "import". [See this keycache demo](https://github.com/wolfSSL/wolfHSM-examples/blob/62ebf7cbbd85951f5fbc58e209a72af2e467bac5/demo/client/wh_demo_client_keystore.c#L120). Maybe we just make the two identical, with the exception of using GCM in this one?

File: demo/client/wh_demo_client_crypto.c [hunk 3]
```diff
@@ -1,7 +1,1347 @@
...
+        goto exit;
+    }
+
+    printf("CMAC hash and verify succeeded\n");
+exit:
+    (void)wc_CmacFree(cmac);
+    return ret;
+}
+
+int wh_DemoClient_CryptoCmacImport(whClientContext* clientContext)
```
Comment (bigbrett) [inline demo/client/wh_demo_client_crypto.c:1262]: same comment re: order of operations - cache first, then write a comment indicating the significance of what we just did and that the remaining cmac operations can just refer to the ID

File: demo/client/wh_demo_client_crypto.c [hunk 4]
```diff
@@ -1,7 +1,1347 @@
...
+    ret = wh_Client_SetKeyIdCmac(cmac, keyId);
+    if (ret != 0) {
+        printf("Failed to wh_Client_SetKeyIdAes %d\n", ret);
+        goto exit;
+    }
+
+    /* verify the tag, note that for pre-cached keys we need to use the special
+     * wolfHSM functions wh_Client_AesCmacGenerate and wh_Client_AesCmacVerify
+     * when doing oneshot cmac generation or oneshot verifition, manual steps
+     * can be done as above */
```
Comment (bigbrett) [inline demo/client/wh_demo_client_crypto.c:1335]:
```suggestion
    /* Verify the tag. Note that in order to use a cached key for 
     * CMAC verification, it is necessary to use the special wolfHSM 
     * client API function: wh_Client_AesCmacVerify() */
```

Comment (bigbrett) [reply demo/client/wh_demo_client_crypto.c:1335]: Also, pedantic to us, but good for the user: we should do a separate one-shot example using the Client API, vs just mentioning it in the comment for the verify

**Review by bigbrett — CHANGES_REQUESTED** [review]
Awesome, thanks so much for this @jpbland1. Just a few changes/cleanups and things to make it clearer for the reader

File: demo/client/wh_demo_client_crypto.c [hunk 5]
```diff
@@ -1,7 +1,1347 @@
...
+    word32 outLen;
+    whKeyId keyIdPriv1 = WOLFHSM_KEYID_ERASED;
+    whKeyId keyIdPub1 = WOLFHSM_KEYID_ERASED;
+    whKeyId keyIdPriv2 = WOLFHSM_KEYID_ERASED;
+    whKeyId keyIdPub2 = WOLFHSM_KEYID_ERASED;
+    /* public-two is the public part of private-one, same for the *-two keys */
+    char privKeyFileOne[] = "../../../demo/certs/curve25519-private-one.raw";
+    char pubKeyFileOne[] = "../../../demo/certs/curve25519-public-one.raw";
+    char privKeyFileTwo[] = "../../../demo/certs/curve25519-private-two.raw";
+    char pubKeyFileTwo[] = "../../../demo/certs/curve25519-public-two.raw";
```
[the remark this answers is not in the document]
Comment (jpbland1) [reply demo/client/wh_demo_client_crypto.c]: great idea

File: demo/client/wh_demo_client_crypto.c [hunk 6]
```diff
@@ -1,7 +1,1347 @@
...
+{
+    int ret;
+    int needEvictPriv;
+    int needEvictPub;
+    word32 outLen;
+    whKeyId keyId = WOLFHSM_KEYID_ERASED;
+    uint8_t sharedOne[CURVE25519_KEYSIZE];
+    uint8_t sharedTwo[CURVE25519_KEYSIZE];
+    curve25519_key curve25519PrivateKey[1];
+    /* public from the first shared secret's perspective, actually private */
```
[the remark this answers is not in the document]
Comment (jpbland1) [reply demo/client/wh_demo_client_crypto.c:192]: it's fine, just named from perspective one

File: demo/client/wh_demo_client_crypto.c (hunk 2 above)
Comment (jpbland1) [reply demo/client/wh_demo_client_crypto.c:1135]: it's fine, the order ultimately doesn't matter so ordering it any way will convey this

File: demo/client/wh_demo_client_crypto.c (hunk 4 above)
Comment (jpbland1) [reply demo/client/wh_demo_client_crypto.c:1334]: I wanted to show the cached key doing operations both manually and with a helper function but I also wanted to make the comment with both functions included so they understand that oneshot generate also has it's own special function

File: demo/client/wh_demo_client_crypto.c [hunk 7]
```diff
@@ -193,7 +193,7 @@ int wh_DemoClient_CryptoCurve25519(whClientContext* clientContext)
     curve25519_key curve25519PublicKey[1];
     WC_RNG rng[1];
 
-    /* initialize rng to make the cruve25516 keys */
+    /* initialize rng to make the cruve25519 keys */
```
Comment (bigbrett) [inline demo/client/wh_demo_client_crypto.c:196]: @jpbland1 Somehow I didn't notice the double typo the first time XD
```suggestion
    /* initialize rng to make the curve25519 keys */
```

File: demo/client/wh_demo_client_crypto.c (hunk 4 above)
Comment (bigbrett) [reply demo/client/wh_demo_client_crypto.c:1334]: I understand, I'm saying go one step further and add a oneshot generate example too, as opposed to just leaving the note about using the client API for oneshots in the comment. Harder to miss. That way we are explicitly demonstrating the correct way to do a oneshot CMAC in the demo code. Just create a `whDemoClient_CryptoCmacOneshotImport()` with the same input data.

**Review by bigbrett — CHANGES_REQUESTED** [review]
missed a typo and think we should create a one shot CMAC demo function, since it is a special case and we should highlight it

**Comment by jpbland1** [comment]
gotcha, added the oneshot example

**Review by bigbrett — APPROVED** [review]

**Pull request merged**